packages/grafana-runtime/src/services/LocationSrv.ts

```diff
@@ -1,36 +1,84 @@
+/**
+ * Passed as options to the {@link LocationSrv} to describe how the automatically navigation
+ * should be performed.
+ *
+ * @public
+ */
 export interface LocationUpdate {
+  /**
+   * Target path where you automatically wants to navigate the user.
+   */
   path?: string;
+
+  /**
+   * Specify this value if you want to add values to the query string of the URL.
+   */
   query?: UrlQueryMap;
 
   /**
-   * Add the query argument to the existing URL
+   * If set to true, the query argument will be added to the existing URL.
    */
   partial?: boolean;
 
   /**
-   * Do not change this unless you are the angular router
+   * Used internally to sync the redux state from Angular to make sure that the redux location
+   * state is in sync when navigating using the Angular router.
+   *
+   * @remarks
+   * Do not change this unless you are the Angular router.
+   *
+   * @internal
    */
   routeParams?: UrlQueryMap;
```

Comment:
note: should probably move this from the public interface and create a separate internal interface to handle the angular sync.

Reply:
Think we can leave this for now. Gonna be gone with React Router migration.